src/ParticleDistributions/ParticleDistributions.jl

```diff
@@ -500,35 +666,68 @@ function update_params_from_moments(ODE_parameters, target_moments::Array{FT}) w
   moments_to_params(dist_prev, target_moments)
 end
 
-function moments_to_params(dist::GammaPrimitiveParticleDistribution{FT}, target_moments::Array{FT}) where {FT<:Real}
+function update_params_from_moments(ODE_parameters, target_moments::Array{FT}, param_range::Dict{String, Tuple{FT, FT}}) where {FT<:Real}
+  # Extract the solution at the previous time step
+  dist_prev = ODE_parameters[:dist]
+  # Update parameters from the given target moments
+  moments_to_params(dist_prev, target_moments; param_range = param_range)
+end
+
+function moments_to_params(dist::GammaPrimitiveParticleDistribution{FT}, target_moments::Array{FT}; param_range = Dict("θ" => (1e-5, 1e5), "k" => (eps(FT), FT(5)))) where {FT<:Real}
   if length(target_moments) != nparams(dist)
     error("Number of moments must be consistent with distribution type.")
   end
-  check_moment_consistency(target_moments)
 
-  # target_moments[1] == M0, target_moments[2] == M1, target_moments[3] == M2
```

Comment:
Do we merge these with update_dist_from_moments in this PR or do we leave it for a future PR?